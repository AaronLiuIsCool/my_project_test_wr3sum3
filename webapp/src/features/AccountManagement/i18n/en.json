{
    "heading": "账号管理",
    "basicInfo": "基本信息",
    "accountManagement": "账号管理",
    "wechat": "微信",
    "email": "邮箱",
    "password": "密码",
    "bind": "前往绑定",
    "unbind": "解除绑定"
  }