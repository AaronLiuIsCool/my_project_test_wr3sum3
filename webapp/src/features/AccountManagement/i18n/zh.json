{
  "heading": "账号管理",
  "basicInfo": "基本信息",
  "accountManagement": "账号管理",
  "wechat": "微信",
  "email": "邮箱",
  "password": "密码",
  "bind": "前往绑定",
  "unbind": "解除绑定",
  "binded": "已绑定",
  "notBinded": "未绑定",
  "reset": "重置",
  "enterEmailAddress": "请输入您所要绑定的邮箱",
  "save": "保存",
  "avatar": "头像",
  "name": "姓名",
  "updateEmail": "更换邮箱",
  "gender": "性别",
  "identity": "身份",
  "genderOptions": [
    {
      "key": "male",
      "value": "男"
    },
    {
      "key": "female",
      "value": "女"
    }
  ],
  "identityOptions": [
    {
      "key": "male",
      "value": "男"
    },
    {
      "key": "female",
      "value": "女"
    }
  ]
}
