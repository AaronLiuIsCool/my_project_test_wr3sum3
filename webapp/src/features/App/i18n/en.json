{
  "brand": "Tailoryourvc",

  "nav_item_resumehub": "Resume Hub",
  "nav_item_jobmatcher": "Jobs",

  "language_toggle_zh": "中文",
  "language_toggle_en": "English",

  "logout": "Sign out",
  "defaultInitial": "Unknown"
}