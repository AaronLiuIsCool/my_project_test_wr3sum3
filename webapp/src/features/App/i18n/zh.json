{
  "brand": "快刀简历",

  "nav_item_resumehub": "我的简历",
  "nav_item_jobmatcher": "工作搜索",
  "nav_item_jobcollction": "我的岗位",
  "language_toggle_zh": "中文",
  "language_toggle_en": "English",
  "createResume": "创建简历",
  "createNewResume": "创建一份全新简历",
  "chooseLanguage": "请选择简历语言",
  "resumeUpload": "我有简历了，我要上传",
  "resumeUploadSupportType": "支持PDF/Word 文档，文件最大支持20MB",
  "uploadYourResume": "从本地上传",
  "upload": "上传",
  "logout": "登出",
  "defaultInitial": "未知",
  "name_it":"请您为这份新简历命名",
  "name_suggestion":"建议用简历类别，比如 [咨询] [投行] [战投] 或者是 [岗位名称+公司名称] 的形式来来命名以方便查询",
  "confirm":"确认",
  "RPreview": {
    "linkedinLink": "领英链接",
    "githubLink": "Github链接",
    "educationBackground": "教育背景",
    "workExperience": "工作经历",
    "projectExperience": "项目经历",
    "studentWorkAndVolunteer": "志愿服务",
    "certificateAndAward": "技能证书 & 获得荣誉",
    "awards": "获得荣誉",
    "certificate": "技能证书",
    "validForever": "永久有效",
    "expiredAt": "到期",
    "current": "现今",
    "editThemeColor": "编辑主题色",
    "smartTranslation": "智能翻译",
    "translationEN": "English",
    "translationCN": "中文版式",
    "oneClickWholePage": "一键整页",
    "downloadResume": "下载简历",
    "perparingResume": "生成中..",
    "downLoadResume": "下载简历",
    "linkedIn": "领英链接",
    "weblink": "项目集链接"
  }
}
