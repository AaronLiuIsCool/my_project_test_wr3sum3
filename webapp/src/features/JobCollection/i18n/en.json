{
    "heading": "我的岗位",
    "all": "全部",
    "applied": "已经申请",
    "postedDate": "发布日期",
    "jobLocation": "岗位所在地",
    "applyJob": "申请工作"
  }