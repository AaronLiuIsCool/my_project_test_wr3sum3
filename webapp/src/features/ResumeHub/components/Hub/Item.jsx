import React from 'react';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from "react-router-dom";

import Button from 'react-bootstrap/Button';
import thumbnail from '../../assets/thumbnail@2x.png'; 
import { ReactComponent as EditIcon } from '../../assets/edit.svg'; 
import { ReactComponent as Download } from '../../assets/download.svg'; 
import { ReactComponent as ShareIcon } from '../../assets/share.svg'; 
import { ReactComponent as DuplicateIcon } from '../../assets/duplicate.svg'; 
import { ReactComponent as DeleteIcon } from '../../assets/delete.svg'; 

import { selectUserId } from 'features/App/slicer';
import { deleteResume } from 'features/App/slicer/account';
import AccountServices from 'shell/services/AccountServices';
import { getLogger } from 'shell/logger';

import styles from '../../styles/Hub.module.css';

const logger = getLogger('ResumeHubItem');
const accountServices = new AccountServices();

const Grid = ({ resume }) => {
    const dispatch = useDispatch();
    const userId = useSelector(selectUserId);

    const handleDelete = async () => {
        dispatch(deleteResume(resume.resumeId));
        try {
            const response = await accountServices.deleteResume(userId, resume);
            const responseJson = await response.json();
            if (!response.success) {
                logger.warn(responseJson.message);
            }
        } catch (exception) {
            logger.error(exception)
        }
    }

    return (
        <div className={styles.itemContainer}>
            <div className={styles.itemContent}>
                <img src={thumbnail} width={160} height={208} alt="Thumbnail" />
                <div className={styles.right}>
                    <div className={styles.name}>
                        {resume.alias}
                    </div>
                    <div className={styles.time}>
                        {new Date(resume.createdAt).toLocaleDateString()}
                    </div>
                    <div className={styles.links}>
                        <Link to={`/resume/${resume.resumeId}`} className={styles.link} >
                            <EditIcon /> 编辑内容
                        </Link>
                        <Button variant="link" className={styles.link}>
                            <Download /> 下载简历
                        </Button>
                        <Button variant="link" className={styles.link}>
                            <ShareIcon /> 链接分享
                        </Button>
                        <Button variant="link" className={styles.link}>
                            <DuplicateIcon /> 复制简历
                        </Button>
                    </div>
                    <Button variant="link" className={styles.delete}
                        onClick={handleDelete} >
                        <DeleteIcon /> 删除
                    </Button>
                </div>
            </div>
            <Button className={styles.matchingBtn}>
                工作匹配
            </Button>
        </div>
    );
}

Grid.propTypes = {
    resume: PropTypes.shape({
        resumeId: PropTypes.string.isRequired,
        alias: PropTypes.string.isRequired,
        thumbnailUri: PropTypes.string.isRequired,
        createdAt: PropTypes.string.isRequired
    })
};

export default Grid;