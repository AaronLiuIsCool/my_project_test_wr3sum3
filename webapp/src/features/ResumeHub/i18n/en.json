{
  "new_resume_description": "You do not have a resume yet, spend five minutes to creat one?",
  "new_resume_start": "Build resume"
}