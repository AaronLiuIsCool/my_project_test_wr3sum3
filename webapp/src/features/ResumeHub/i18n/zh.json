{
  "new_resume_description": "看似您还没有简历，花5分钟创造一份吧",
  "new_resume_start": "开始创建",

  "hub_title": "我的简历",
  "hub_action": "+ 添加新简历",

  "hub_item_edit": "编辑内容",
  "hub_item_download": "下载简历",
  "hub_item_share": "链接分享",
  "hub_item_duplicate": "复制简历",
  "hub_item_delete": "删除",
  "hub_item_match": "工作匹配"
}