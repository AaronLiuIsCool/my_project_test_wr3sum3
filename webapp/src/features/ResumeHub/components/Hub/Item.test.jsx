import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { I8nContext } from 'shell/i18n';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import 'react-dates/initialize';

import store from 'store';
import Item from './Item';

const ItemWithProvider = (props) => (
  <Provider store={store}>
    <I8nContext.Provider value={{}}>
      <Item {...props} />
    </I8nContext.Provider>
  </Provider>
);
jest.mock('moment', () => () => ({format: () => '2018–01–30T12:34:56+00:00'}));

describe('Item', () => {
  const showEditIcon = jest.fn();
  test('should render successfully with input', () => {
    const wrapper = mount(
      <Router history={createMemoryHistory()}>
        <ItemWithProvider
          resume={{
            alias: '123',
            resumeId: '123321',
            thumbnailUri: 'url',
            createdAt: '2020-11-11',
          }}
          messages={{}}
          index={0}
        />
      </Router>
    );
    console.log(wrapper.debug());
    expect(wrapper.find('Item').length).toEqual(1);
  });
  
  test('It should match snapshot', () => {
    const tree = renderer
      .create(
        <Router history={createMemoryHistory()}>
          <ItemWithProvider
            resume={{
              alias: '123',
              resumeId: '123321',
              thumbnailUri: 'url',
              createdAt: '2020-11-11',
            }}
            messages={{}}
            index={0}
          />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
