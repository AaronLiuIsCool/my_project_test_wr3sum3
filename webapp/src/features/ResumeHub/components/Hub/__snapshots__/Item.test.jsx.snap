// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item It should match snapshot 1`] = `
<div
  className="itemContainer"
>
  <div
    className="itemContent"
  >
    <div
      className="right"
    >
      <div
        className="aliasContainer"
      >
        <input
          className="name"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          value=""
        />
        <div
          className="edit"
        >
          <img
            alt="edit"
            src="edit.svg"
          />
        </div>
      </div>
      <div
        className="time"
      >
        11/10/2020, 4:00:00 PM
      </div>
      <div
        className="links"
      >
        <a
          className="link btn btn-link"
          disabled={false}
          href="/resume/123321"
          onClick={[Function]}
        >
          <svg
            className="svg"
          >
            edit.svg
          </svg>
        </a>
        <button
          className="link btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            className="svg"
          >
            download.svg
          </svg>
        </button>
      </div>
      <button
        className="delete btn btn-link"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          className="svg"
        >
          delete.svg
        </svg>
         
      </button>
    </div>
  </div>
  <a
    className="matchingBtn btn btn-primary"
    disabled={false}
    href="/jobs?resume=123321"
    onClick={[Function]}
  />
</div>
`;
